src/mapgen/mg_biome.cpp

```diff
@@ -262,6 +263,30 @@ Biome *BiomeGenOriginal::calcBiomeFromNoise(float heat, float humidity, v3s16 po
 		}
 	}
 
+	// If we have a pointer to a place to store the next biomes largest Y then we get and store it
+	if (min_y) {
+		*min_y = pos.Y;
+
+		// Doing a check to see if there is a biome that is hidden in the current biome
+		for (size_t i = 1; i < m_bmgr->getNumObjects(); i++) {
+			Biome *b = (Biome *)m_bmgr->getRaw(i);
+			if (!b ||
+					pos.Y < b->max_pos.Y ||
```

Comment:
This does not seem to be quite right.
`dist_min` (condition below) is only updated when `pos.Y <= b->max_pos.Y + b->vertical_blend`. This means that only a thickness of `b->vertical_blend` is considered. If I am seeing this correctly, this function is executed for almost every `y` position because `y < biome_y_min` always applies. This works but is inefficient.

I tried to implement an alternative by moving this check into the loop above but I found that it might be best to use the same strategy as `MapgenBasic::generateBiomes`: by `biome_y_min = biome->min_pos.Y;`.In that case, neither would take vertical blending into account but it should fix this issue with minimal changes and keeps it consistent.

Reply:
Should I go back to https://github.com/minetest/minetest/pull/13472/commits/781cf4b6298edc42f3cc8aa5ba206bb410a8126a? Basically were you reviewed before.

I don't think there is a more efficient way to calculate biome blending with how the generation currently works.

The current way I think I set it up is that it gets the biome for every y value in a blending zone because it can't know at any point what biome that will be ahead of time due to RNG. Outside of the blending zones it should just keep going using the current biome until it hits the next zone.

Reply:
That commit does not work 100% reliably either because it also takes biomes into account that have different heat/humidity points.
As for efficiency.. the following modification in `MapgenBasic::generateBiomes` could be used to confirm whether the Y limit is set correctly:
```C++
				// (Re)calculate biome
				biome = biomegen->getBiomeAtIndex(index, v3s16(x, y, z), &biome_y_min);
				if (x == 0 && z == 0)
					printf("check @ %i -> %s -> again at %i\n", y, biome->name.c_str(), biome_y_min);
```


I tried the following mod which should result in a stone | 6000x dirt | 10x stone |10x ice | stone sandwich; so far this only happens when the biome is evaluated for every position:
```Lua
minetest.clear_registered_biomes()

minetest.register_biome({
	name = "dirt_plate",
	node_stone  = "basenodes:dirt",
	y_max = -5000,
	y_min = -11000,
	vertical_blend = 0,
	heat_point = 50,
	humidity_point = 50,
})

minetest.register_biome({
	name = "dirt_plate_2",
	node_stone  = "basenodes:ice",
	y_max = -11010,
	y_min = -11020,
	vertical_blend = 0,
	heat_point = 50,
	humidity_point = 50,
})
```